synapse/handlers/sync.py

```diff
@@ -814,9 +845,12 @@ def get_lazy_loaded_members_cache(self, cache_key):
                 # about them).
                 state_filter = StateFilter.all()
 
-                state_at_previous_sync = await self.get_state_at(
-                    room_id, stream_position=since_token, state_filter=state_filter
-                )
+                if since_token:
+                    state_at_previous_sync = await self.get_state_at(
+                        room_id, stream_position=since_token, state_filter=state_filter
+                    )
+                else:
+                    state_at_previous_sync = {}
```

Comment:
this looks odd; what's going on?

Reply:
Ah, yes, sorry. This is what other code paths do in the function does. I sort of assume that there is a reason why we never end up in this particular code path in an initial sync, but it wasn't obvious if that was the case (`get_state_at` will explode if given a `None` since token). So either a) the branch is never hit but at least the typing is correct or b) this is fixing an obscure bug with certain initial sync requests.

Reply:
kinda wonder if we ought to `assert` it rather than have potentially-unreachable code. I could go either way though.

Reply:
I'd be more comfortable asserting if I understood *why* its unreachable.

Reply:
Actually, looking at the code again that path is when `full_state is False`, which I guess is means that it's not an initial sync?

Reply:
Changed this to an assert.